src/commonMain/kotlin/fr/acinq/lightning/wire/MessageOnion.kt

```diff
@@ -58,6 +58,97 @@ sealed class OnionMessagePayloadTlv : Tlv {
                 EncryptedData(ByteVector(LightningCodecs.bytes(input, input.availableBytes)))
         }
     }
+
+    /**
+     * In order to pay a Bolt 12 offer, we must send an onion message to request an invoice corresponding to that offer.
+     * The creator of the offer will send us an invoice back through our blinded reply path.
+     */
+    data class InvoiceRequest(val tlvs: TlvStream<OfferTypes.InvoiceRequestTlv>) : OnionMessagePayloadTlv() {
+        override val tag: Long get() = InvoiceRequest.tag
+        override fun write(out: Output) = OfferTypes.InvoiceRequest.tlvSerializer.write(tlvs, out)
+
+        companion object : TlvValueReader<InvoiceRequest> {
+            const val tag: Long = 64
+
+            override fun read(input: Input): InvoiceRequest =
+                InvoiceRequest(OfferTypes.InvoiceRequest.tlvSerializer.read(input))
+        }
+    }
+
+    /**
+     * When receiving an invoice request, we must send an onion message back containing an invoice corresponding to the
+     * requested offer (if it was an offer we published).
+     */
+    data class Invoice(val tlvs: TlvStream<OfferTypes.InvoiceTlv>) : OnionMessagePayloadTlv() {
+        override val tag: Long get() = Invoice.tag
+        override fun write(out: Output) = tlvSerializer.write(tlvs, out)
+
+        companion object : TlvValueReader<Invoice> {
+            const val tag: Long = 66
+
+            val tlvSerializer = TlvStreamSerializer(
```

Comment:
I think it would be more consistent to have this in `OfferTypes`, like we do for `Offer` and `InvoiceRequest`.